### ISSUE TITLE
prefer-arrow-callback and named functions

### ISSUE THREAD

**Issue by zertosh** [eslint/eslint#3454]
`prefer-arrow-callback` is incorrectly warning when using a named callback. If the function is named and it's being referenced, it should not warn because arrow functions can only be anonymous. Example:

```js
// warns but it shouldn't
foo(function bar () {
	bar();
});
```

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by mysticatea** [comment]
Thanks for the report.

You are right, I will fix this.

**Comment by vtfn** [comment]
I know this issue has already been closed, but I am still having this problem, even after #3461

**Comment by mysticatea** [comment]
@vtfn Could you share steps to reproduce it?

**Comment by vtfn** [comment]
@mysticatea 

function foo(fn) {
  fn();
}

foo(function named() {
  console.log('this is just a useless function example');
});

Even though the 'named' function isn't been called from inside itself, the code above should not raise a warning.
Using named functions like this improve stack traces.

**Comment by mysticatea** [comment]
@vtfn It's not the intention of this issue, so please open an issue.

**Comment by vtfn** [comment]
Ok. sorry.
